### ISSUE TITLE
Please add keyboard shortcut support!

### ISSUE THREAD

**Issue by Hlsgs** [gyng/save-in#15]
(self explaining)

**Comment by gyng** [comment]
This would be really nice. For Firefox, it's blocked by https://bugzilla.mozilla.org/show_bug.cgi?id=1320462 
From that page it seems like Chrome has an undocumented feature for it. Think I'll wait for Firefox to implement it too to save me some pain later on.

**Comment by Hlsgs** [comment]
How about SHIFT + doubleclick as a workaround?

**Comment by gyng** [comment]
Menu shortcuts landed for Chrome in 2.0.0. Right-click > a > a or Right-click > a > {1, 2, 3...} by default.

Directly clicking on the item will require injecting JS into the page, which I want to avoid if possible. Maybe if Firefox takes forever to implement accesskeys.

**Comment by gyng** [comment]
It's the other bug, I think?

I've changed my mind on content scripts as there's not much movement on the linked bugs. In 2.2.0 there's now an experimental option for clicking to save. It's disabled by default and the shortcut defaults to Alt+click. A different modifier key can be supplied.

**Comment by FatOrangutan** [comment]
Is it possible to prevent the click from propagating through when used as part of the click-to-save feature?

Current behaviour: alt+click also zooms in / out image etc.
Expected behaviour: consume the click.

**Comment by Sjkomy** [comment]
gyng, So where is that option exactly? Can't find it.

**Comment by gyng** [comment]
@FatOrangutan Released 2.2.1 that hopefully fixes this.

@Sjkomy It's under keyboard shortcuts in 2.2.0
![screen shot 2017-12-27 at 10 08 39](https://user-images.githubusercontent.com/370496/34368762-5c9c8214-eaf1-11e7-8761-edac00ca7dbc.png)

**Comment by Sjkomy** [comment]
Perfect! It seems that it updated itself overnight otherwise i couldn't miss that EXPERIMENTAL button. Thank you, i almost feel as comfy as i was before that firefox update, and if only there was a context menu editor...

**Comment by GallaherMike** [comment]
So this doesn't seem to be working for me. Does the injection still happen in the following situation.

Page of thumbnail links to full size images, open several in new tabs. These new tabs have the full sized images only, JPG, PNG, etc... not embedded in an HTML page. Does the script injection still work in this case? When I go to those tabs try to ALT+Click nothing happens.

It would also be nice if this could also be driven by a Double Click, but the ALT+Click is a great start.

**Comment by gyng** [comment]
@GallaherMike I can't replicate this. It works, but the individual tabs have to be refreshed for the script injection to work if the setting was changed *after* the tabs were loaded.

**Comment by Sjkomy** [comment]
@gyng I spotted a bug that sometimes after i open a new page every image i press is downloaded (without hotkey), refresh undoes this

**Comment by gyng** [comment]
@Sjkomy Thanks, I've noticed this too. Should be fixed for the next release in cbd519b703bcc616f7f18da50b3bf5da0d4836e4

**Comment by gyng** [comment]
@Pandahisham It's available as accesskeys for Chrome, but will be nice to have something equivalent for Firefox. I'll look to add it in some form.

**Comment by Pandahisham** [comment]
@gyng I think I failed to explain myself properly. The accesskeys still require me to manually click and open the context menu first , then hit the accesskeys . What I am proposing is allowing multiple "click to save" keycodes, each bound to a specific folder so that the user need not open the context menu. Thanks for all your effort
